# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLStyleEditor.cpp
# Commit: 20512741e7b6
# Full Hash: 20512741e7b60f3f6791002d8311d681f9242e68
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2023-02-02 09:15:42
# Regressor Bug: 1792654
# File Overlap Count: 4
# Description:
#   Bug 1812170 - Make `HTMLEditor::ClearStyleAt` should return `pointToPutCaret` if next node of first split is not split r=m_kato
#   
#   The new path added in bug 1807829 hits the odd result case of `ClearStyleAt`.
#   `pointToPutCaret` may be updated if the preceding split occurs.  Therefore,
#   it should be returned if it does not split next nodes after the first split
# ==============================================================================

diff -r d9138ddf3c3d -r 20512741e7b6 editor/libeditor/HTMLStyleEditor.cpp
--- a/editor/libeditor/HTMLStyleEditor.cpp	Thu Feb 02 02:25:04 2023 +0000
+++ b/editor/libeditor/HTMLStyleEditor.cpp	Thu Feb 02 05:09:15 2023 +0000
@@ -2373,11 +2373,14 @@
     }
   }
 
+  if (!unwrappedSplitResultAtStartOfNextNode.Handled()) {
+    return std::move(pointToPutCaret);
+  }
+
   // If there is no content, we should return here.
   // XXX Is this possible case without mutation event listener?
-  if (NS_WARN_IF(!unwrappedSplitResultAtStartOfNextNode.Handled()) ||
-      !unwrappedSplitResultAtStartOfNextNode.GetPreviousContent()) {
-    // XXX This is really odd, but we retrun this value...
+  if (!unwrappedSplitResultAtStartOfNextNode.GetPreviousContent()) {
+    // XXX This is really odd, but we return this value...
     const auto splitPoint =
         unwrappedSplitNodeResult.AtSplitPoint<EditorRawDOMPoint>();
     const auto splitPointAtStartOfNextNode =